azure-pipelines-docs-integration.yml

```diff
@@ -55,6 +55,15 @@ stages:
           - bash: ./scripts/check_for_docs_deps_changes
             name: CheckDocsDependenciesChanges
 
+  - stage: custom_checks
+    pool:
+      vmImage: 'ubuntu-latest'
+    jobs:
+    - job: link_checker
+      condition: or(eq(stageDependencies.scope_check.changes.outputs['CheckDocsChanges.DocsChanged'], true), eq(variables.isMain, true))
+      steps:
+      - bash: python scripts/docs_link_checker.py -p docs -r docs -s docs --skip-external
+
```

Comment:
might we want to add a `name` here so it doesn't show up as `Bash` in CI/CD? I'm good either way.